### PR TITLE
Upgrade typescript@3.5.3

### PR DESCRIPTION
**Pull request by markuskobler**
Fix's #78

### CODE REVIEW

**Comment by asakusuma** [comment]
@krisselden can you review this?

File: src/compiler/create-parse-config-host.ts [hunk 1]
```diff
@@ -11,6 +11,14 @@ export default function createParseConfigHost(
     return input.getFileSystemEntries(path);
   }
 
+  function realpath(path: string): string  {
+    try {
+      return input.realpath(path) || path;
+    } catch {
+      return path;
```
Comment (krisselden) [inline src/compiler/create-parse-config-host.ts]: need to throw if not the expected error `if (e.code !== "ENOENT") { throw e; }`

File: src/compiler/create-parse-config-host.ts [hunk 2]
```diff
@@ -11,6 +11,14 @@ export default function createParseConfigHost(
     return input.getFileSystemEntries(path);
   }
 
+  function realpath(path: string): string  {
+    try {
+      return input.realpath(path) || path;
```
Comment (krisselden) [inline src/compiler/create-parse-config-host.ts]: Is the `|| path` really needed in addition to the `try/catch`? There is no explanation of the failure scenario here.

**Review by krisselden — DISMISSED** [review]
Need more explanation to how this is fixes #78 where @asakusuma mentioned that matchFiles changed, I expected some change related to that.

**Comment by wycats** [comment]
Can we merge this so we can get up to date?

**Comment by wycats** [comment]
It seems like TS thinks we're using private APIs? 

https://github.com/microsoft/TypeScript/pull/32564

**Comment by krisselden** [comment]
We need to, it is too cumbersome to match the globbing behavior in the host interfaces without using that API, so that is not going to change.

**Comment by krisselden** [comment]
I've asked them to make it public or easier.  It should be as easy to make a virtual fs like they use in their test suite with public api.

**Comment by krisselden** [comment]
linking some related issues

https://github.com/microsoft/TypeScript/issues/13793
https://github.com/microsoft/TypeScript/issues/20878
https://github.com/microsoft/TypeScript/pull/21021

There is no reasonable way to deal with include/exclude globing in readDirectory in a host that is trying to gate all fs access without this API.

Broccoli treats input as immutable and actually builds inside of tmp directories, the only way to get node resolution and types to work is to pretend in the host like these tmp dirs are "mounted" in the project path.  In order to avoid using this API, I would have to parse the glob and match the under-specified glob behavior of matchFiles.

**Pull request merged**